soc: arm: microchip: add common mec_gpio.h

jvasanth1 opened the pull request:
add common mec_gpio.h to allow access to common gpio_regs structure for both mec15xx and mec17xx, used in pinctrl driver

stephanosio commented on drivers/pinctrl/pinctrl_mchp_xec.c:
```diff
@@ -12,7 +12,7 @@
 
 /** @brief All GPIO register as arrays of registers */
-struct gpio_regs {
+struct gpio_regs_t {
```
> so that it doesn't confuse with gpio_regs structure already defined in mec172x_gpio.h in soc

then why is it redefined here?

jvasanth1 replied:
the pinctrl driver was updated to work with both mec15xx and mec172x and in that process the common definitions were moved to the pinctrl source file. Here is the related [PR 47971](https://github.com/zephyrproject-rtos/zephyr/pull/47971)

gmarull replied:
Does that mean we include unnecessary stuff?

jvasanth1 replied:
okay, I will work to move the common stuff to `soc/arm/microchip_mec/common/reg/ `folder

jvasanth1 replied:
moved the common `gpio_regs` structure to `soc/arm/microchip_mec/common/reg/ folder`
updated pinctrl driver accordingly

jvasanth1 committed:
soc: arm: microchip: add common mec_gpio.h

add common mec_gpio.h to allow access to common gpio_regs
structure for both mec15xx and mec17xx, used in pinctrl driver

Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

MaureenHelm commented on drivers/pinctrl/pinctrl_mchp_xec.c:
```diff
@@ -10,142 +10,7 @@
-	 MCHP_GPIO_CTRL_MUX_GPIO + MCHP_GPIO_CTRL_OUTV_HI)
-
-/*
- * Each GPIO pin implements a second control register.
- * GPIO Control 2 register selects pin drive strength and slew rate.
- * bit[0] = slew rate: 0=slow, 1=fast
- * bits[5:4] = drive strength
- * 00b = 2mA (default)
- * 01b = 4mA
- * 10b = 8mA
- * 11b = 12mA
- */
-#define MCHP_GPIO_CTRL2_OFFSET		0x0500u
-#define MCHP_GPIO_CTRL2_SLEW_POS	0
-#define MCHP_GPIO_CTRL2_SLEW_MASK	0x01u
-#define MCHP_GPIO_CTRL2_SLEW_SLOW	0
-#define MCHP_GPIO_CTRL2_SLEW_FAST	BIT(MCHP_GPIO_CTRL2_SLEW_POS)
-#define MCHP_GPIO_CTRL2_DRV_STR_POS	4
-#define MCHP_GPIO_CTRL2_DRV_STR_MASK	0x30u
-#define MCHP_GPIO_CTRL2_DRV_STR_2MA	0
-#define MCHP_GPIO_CTRL2_DRV_STR_4MA	0x10u
-#define MCHP_GPIO_CTRL2_DRV_STR_8MA	0x20u
-#define MCHP_GPIO_CTRL2_DRV_STR_12MA	0x30u
-
-#define MCHP_XEC_NO_PULL	0x0
-#define MCHP_XEC_PULL_UP	0x1
-#define MCHP_XEC_PULL_DOWN	0x2
-#define MCHP_XEC_REPEATER	0x3
-#define MCHP_XEC_PUSH_PULL	0x0
-#define MCHP_XEC_OPEN_DRAIN	0x1
-#define MCHP_XEC_NO_OVAL	0x0
-#define MCHP_XEC_OVAL_LOW	0x1
-#define MCHP_XEC_OVAL_HIGH	0x2
-#define MCHP_XEC_DRVSTR_NONE	0x0
-#define MCHP_XEC_DRVSTR_2MA	0x1
-#define MCHP_XEC_DRVSTR_4MA	0x2
-#define MCHP_XEC_DRVSTR_8MA	0x3
-#define MCHP_XEC_DRVSTR_12MA	0x4
-
-#define SHLU32(v, n)			((uint32_t)(v) << (n))
```
None of these defines are being used?

jvasanth1 replied:
some definitions are being used
with the inclusion of <soc.h> the gpio header file in the soc folder has the definitions

jvasanth1 commented:
@MaureenHelm can you help to review and approve?
